src/Trakx.CoinGecko.ApiClient/CoinGeckoClients/CoinGeckoClient.cs

```diff
@@ -27,11 +27,13 @@ public CoinGeckoClient(
         _simpleClient = simpleClient;
         _searchClient = searchClient;
         _dateTimeProvider = dateTimeProvider;
-        _typeName = GetType().FullName;
+        _typeName = GetType().Name;
     }
 
-    private async Task<T> GetFromCacheOrApi<T>(string cacheKey, Func<Task<T>> getFromApi)
+    private async Task<T> GetFromCacheOrApi<T>(object?[] keyFragments, Func<Task<T>> getFromApi)
```

Comment:
[suggestion] I would suggest to leave the method as 
`private async Task<T> GetFromCacheOrApi<T>(string cacheKey, Func<Task<T>> getFromApi)`

and the caller calls 
`string BuildCacheKey(params object?[] keyFragments)`

Reply:
That makes a lot of sense, thank you, done.